### PR TITLE
add option to forward headers 

### PR DESCRIPTION
**Pull request by zuzana**
I have a custom set prerender and I would need to forward special headers from SPA to prerender for testing purposes.

Also there's similar issue about [forwarding Accept-Language header](https://github.com/prerender/prerender/issues/196).

Let me know, what you think. Thanks.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -163,10 +163,13 @@ prerender.getPrerenderedPageResponse = function(req, callback) {
     uri: url.parse(prerender.buildApiUrl(req)),
     followRedirect: false
   };
-  options.headers = {
-    'User-Agent': req.headers['user-agent'],
-    'Accept-Encoding': 'gzip'
-  };
+  if (this.forwardHeaders === true) {
+    Object.keys(req).forEach(function(h) {
```
Comment (thoop) [inline index.js]: Shouldn't this be `Object.keys(req.headers)`?

**Comment by thoop** [comment]
Tests are failing so I can't merge until that's fixed.

You also might want to default headers before setting them:
```
var options = {
    uri: url.parse(prerender.buildApiUrl(req)),
    followRedirect: false,
    headers: {}
  };
```

**Comment by zuzana** [comment]
tests fixed, thanks for review

**Comment by zuzana** [comment]
@thoop: Have you had some time to look at it again? Thanks.

**Comment by thoop** [comment]
Not yet. Been very busy. Sorry!

**Comment by thoop** [comment]
Thanks!

**Pull request merged**